test/unit/gateways/stripe_payment_intents_test.rb

```diff
@@ -259,6 +259,20 @@ def test_failed_payment_methods_post
     assert_equal 'invalid_request_error', create.params.dig('error', 'type')
   end
 
+  def test_invalid_login_test_transaction
+    gateway = StripePaymentIntentsGateway.new(login: 'sk_live_3422')
+    assert response = gateway.purchase(@amount, @credit_card, @options)
+    assert_failure response
+    assert_match 'Invalid API Key provided', response.message
+  end
+
+  def test_invalid_login_live_transaction
+    gateway = StripePaymentIntentsGateway.new(login: 'sk_test_3422', test: false)
```

Comment:
This may be my morning brain but how are we doing a check for `production` gateways? I see `test? && !key(options).include?('sk_test')` and to me that reads that it only validates if `test=true` but here you're saying we validate also when `test=false`?

Reply:
You are right, I forgot that part which I realized after I created the PR lol I'll add the production check